### PR TITLE
Fix Authenticator Maps

### PR DESCRIPTION
**Pull request by slemrmartin**
Fixes URL and validations for AuthenticatorMap.option

`order` validation doesn't need to be present if there is default value (that is not compatible with the way how ansible collection can work)
`triggers` validation also doesn't need to be present in PATCH request

### CODE REVIEW

File: ansible_base/authentication/serializers/authenticator_map.py [hunk 1]
```diff
@@ -27,14 +24,30 @@ def validate(self, data) -> dict:
...
-
         if errors:
             raise ValidationError(errors)
         return data
 
-    def validate_trigger_data(self, triggers: dict, definition, error_prefix: str) -> dict:
+    def validate_trigger_data(self, data):
+        errors = {}
+        request = self.context.get('request', None)
+        if not request or (request.method != 'PATCH'):
```
Comment (AlanCoding) [inline ansible_base/authentication/serializers/authenticator_map.py]: It's difficult to imagine that `_validate_trigger_data` should be called differently when changed in PATCH vs PUT.

Comment (AlanCoding) [reply ansible_base/authentication/serializers/authenticator_map.py]: Like, if it is a PATCH and "triggers" is in data, then you should call `_validate_trigger_data`.

Comment (slemrmartin) [reply ansible_base/authentication/serializers/authenticator_map.py]: nice catch, yeah.

Comment (slemrmartin) [reply ansible_base/authentication/serializers/authenticator_map.py]: switched conditions, now it should be ok

File: ansible_base/authentication/serializers/authenticator_map.py [hunk 2]
```diff
@@ -27,14 +24,30 @@ def validate(self, data) -> dict:
...
             raise ValidationError(errors)
         return data
 
-    def validate_trigger_data(self, triggers: dict, definition, error_prefix: str) -> dict:
+    def validate_trigger_data(self, data):
+        errors = {}
+        request = self.context.get('request', None)
+        if 'triggers' not in data or not data['triggers']:
+            if not request or (request.method != 'PATCH'):
+                errors["triggers"] = "Triggers must be a valid dict"
```
Comment (AlanCoding) [inline ansible_base/authentication/serializers/authenticator_map.py:36]: Final drive-by comment about this

https://github.com/ansible/django-ansible-base/blob/5babe8742e57d45ea703986398c2777fb370f991/ansible_base/authentication/models/authenticator_map.py#L64-L69

If this shouldn't be left out of create data, or set to `{}` (the default) by omitting from PUT data, then what you're really asking for starts to sound like `blank=False`.

Comment (slemrmartin) [reply ansible_base/authentication/serializers/authenticator_map.py:36]: that's true, the question is, if it was the original intention that empty dict is not possible. 

@john-westcott-iv do you know more about this?

Comment (slemrmartin) [reply ansible_base/authentication/serializers/authenticator_map.py:36]: @AlanCoding ... the intention is not to allow `{}`. I tried `blank=False`, which doesn't work, it's not equal to this validation. So it has to stay here

**Review by AlanCoding — APPROVED** [review]
